homeassistant/components/conversation/__init__.py

```diff
@@ -210,16 +210,32 @@ async def _async_converse(
 ) -> intent.IntentResponse:
     """Process text and get intent."""
     agent = await _get_agent(hass)
+    if language is None:
+        language = hass.config.language
+
     try:
         intent_result = await agent.async_process(
             text, context, conversation_id, language
         )
     except intent.IntentHandleError as err:
+        # Match was successful, but target(s) were invalid
+        intent_result = intent.IntentResponse(language=language)
+        intent_result.async_set_error(
+            intent.IntentResponseErrorReason.NO_VALID_TARGETS, str(err)
+        )
```

Comment:
For a future PR, we should allow intent errors to be converted to intent result. That way the code raising the error can fill in all the relevant details (like missing entities etc).
```suggestion
        intent_result = err.to_intent_response()
```